src/Razor/test/VSCode.FunctionalTest/tests/index.ts

```diff
@@ -0,0 +1,103 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as fs from 'fs';
+import * as Mocha from 'mocha';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+// This file controls which tests are run during the functional test process.
+export async function run(): Promise<void> {
+    const mocha = new Mocha({
+        ui: 'tdd',
+        timeout: 60000,
+    });
+    mocha.useColors(true);
+
+    const testsRoot = path.resolve(__dirname, '..');
+    const testArtifacts = path.join(testsRoot, '..', '..', '..', '..', 'artifacts', 'TestResults');
+    ensureDirectory(testArtifacts);
+    const config = process.env.config ? process.env.config : 'Debug';
+    const testResults = path.join(testArtifacts, config);
+    ensureDirectory(testResults);
+    const resolvedTestResults = path.resolve(testResults);
+    ensureDirectory(resolvedTestResults);
+    const file = path.join(resolvedTestResults, 'VSCode-FunctionalTests.xml');
+    mocha.reporter('xunit', { output: file });
+
+    const razorConfiguration = vscode.workspace.getConfiguration('razor');
+    const devmode = razorConfiguration.get('devmode');
+    if (!devmode) {
```

Comment:
Would it be valuable to move this check up earlier in the method?

Reply:
Sure. It'd put all mocha stuff together